htdocs/xoops_lib/Xoops/Core/Logger.php

```diff
@@ -204,7 +204,7 @@ private function reportFatalError($msg)
         if ('cli' === php_sapi_name()) {
             fprintf(STDERR, "\nError : %s\n", $msg);
         } else {
-            printf(_XOOPS_FATAL_MESSAGE, XOOPS_URL, $msg);
+            printf(_XOOPS_FATAL_MESSAGE, XOOPS_URL, utf8_encode($msg));
```

Comment:
Interesting... now I am curious. The message doesn't come from XOOPS. If the initial DBAL connection attempt throws an exception, we will give an untranslated (at the moment) message. It looks like Doctrine DBAL would use what came from PDO::errorInfo(), which might change based on what layer breaks. So, I'm not certain if this is coming from PDO or MySQL code.  The more I look for answers, the more I find unanswered questions :(

It seems certain we can't really count on the message language or character set, so we can make a best effort to make it better. We've got a library [patchwork/utf8](https://github.com/tchwork/utf8), that has a method made for this. We already use that to make sure our input is well formed, so it would be consistent.

How about we change it to:
```PHP
printf(_XOOPS_FATAL_MESSAGE, XOOPS_URL, Patchwork\Utf8::filter($msg));
```